shared_vcx/src/maybe_known.rs

```diff
@@ -0,0 +1,13 @@
+use serde::{Deserialize, Serialize};
+
+/// Enum used to encapsulate `string-like` types which may
+/// have variants we haven't implement yet.
+///
+/// Deserialization will be first attempted to the [`MaybeKnown::Known`] variant
+/// and then, if that fails, to the [`MaybeKnown::Unknown`] variant.
```

Comment:
The `string-like` thing is the default, but not the only way to use this. The comment should be updated to reflect that.